### PR TITLE
nixos/ebusd: fix device access

### PR DESCRIPTION
**Pull request by Birdy2014**
When `services.ebusd.device` is set to a local device like `/dev/ttyACM0`, ebusd will fail with the error `device invalid`. This is because the module disallows all device access using `PrivateDevices` when a device is specified, instead of allowing device access.

I only tested this change with an adapter that is connected via USB, but this shouldn't change the behavior of adapters connected over the network.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Birdy2014** [comment]
@nathan-gs

File: nixos/modules/services/home-automation/ebusd.nix
Comment (FliegendeWurst) [inline nixos/modules/services/home-automation/ebusd.nix:172]: What if the device is ens:DEVICE or enh:DEVICE? Does that usage already work with PrivateDevices = true?

Comment (Birdy2014) [reply nixos/modules/services/home-automation/ebusd.nix:172]: Oh, I missed that. I made some changes and it also works now with ens:DEVICE and enh:DEVICE.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by Birdy2014** [comment]
This should also be backported to 24.11. Do I need to do anything for that?

**Comment by FliegendeWurst** [comment]
> This should also be backported to 24.11. Do I need to do anything for that?

I added the label, it will be backported once merged. (You can add labels on your own if you add yourself to the maintainer list.)

**Comment by FliegendeWurst** [comment]
Please fix the merge conflict. For some reason the file was reformatted as part of #322537 even though it was updated in this recent PR.

**Comment by FliegendeWurst** [comment]
I think you dropped the PrivateDevices change accidentally.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**
